### PR TITLE
Add failing test for class-level esproposal_decorators

### PR DESCRIPTION
**Pull request by idolize**
The `esproposal_decorators` option does not currently include decorators applied to the class itself.

This change includes a **FAILING test** of class-level decorators to ensure proper support is added.

### CODE REVIEW

**Comment by jeffmo** [comment]
We didn't include this syntax in the flag because it's still not clear if it will be part of the actual proposal. There has been discussion about it, but no clear decisions have been made about how to address the potential caveats involved. Specifically: (1) That it would be possible to decorate a class declaration, but not a function decoration and (2) That toplevel decorators would require an entirely different param/return signature than "property" signatures currently do.

So until there's further discussion that makes it clear that this is likely to move forward with the rest of the decorators proposal we should at least gate this behind a separate flag.

Moreover, if we go the route of enabling this (even if behind a separate flag) we need to add support for it to the parser and inference code in addition to adding it to a test.

cc @wycats to chime in on thoughts about a sub-flag

**Comment by idolize** [comment]
@jeffmo Thanks for the response! I realize this PR is woefully incomplete, but I wasn't aware of the contention surrounding class-based decorators vs. function decorators, so thanks for elaborating on that!

As an aside: I totally understand the reasoning to hold off on adding this as a feature given the long-term uncertainty, but the JS (and specifically React) community is already utilizing decorators on class declarations pretty heavily at the moment.

So, being pragmatic about it, having some kind of simple escape hatch here would go a *long way* towards unblocking use of flow with existing ES7 + babel codebases in the short term. (Especially given that [right now we can't even use `suppress_comment`s](https://github.com/facebook/flow/issues/916#issuecomment-146563896) to stop the flood of errors).

**Comment by jeffmo** [comment]
Totally! I'm not opposed to adding an ignore flag for it like property decorators -- I just want to get a grip on how likely it is to ship with the existing proposal vs defer to a secondary/additive proposal (and thus deserve its own flag).

As for suppress_comments not suppressing parse errors -- that's certainly a known bug that we need to address regardless of what we do here. (cc @gabelevi)

**Comment by stephanos** [comment]
@idolize are you working on the ignore flag for class decorators or did you find another workaround?

**Comment by idolize** [comment]
@stephanos Neither, unfortunately. I don't know enough OCaml to write it myself, and I was unable to find a suitable workaround in the meantime.

**Comment by STRML** [comment]
As decorators are coming into wider use in React projects (Redux uses it everywhere, and users still on mixins use it with react-mixin), it would be really great to have at least an ignore flag for this.

**Comment by cheapsteak** [comment]
Are there any updates on this, or has anyone found a workaround?
Was hoping to use Flow on our new project but not being able to use decorators is a fairly large deal

**Comment by vjpr** [comment]
@cheapsteak See #606. Its ugly but I've found its not too bad, and I've bound it to a surround live template in IntelliJ.

And Flow is working really well for me now, and catching tons of little bugs :)

```
/*::`*/ @deco /*::`*/
class Foo { ...
```

**Comment by samwgoldman** [comment]
Parsing support for decorators is absolutely on my radar. Clearly the community has decided to adopt this, so I think it's a reasonable time commitment to build in parsing support.

This PR is in a weird position, because we can't accept it—it will break the build. I'm going to close this for that reason. On the bright side, the change here is quite simple, and we will certainly add this back in once we add parsing support.

TL;DR, message received. This is on the list. Thanks for providing your feedback, everyone.

**Comment by tomatau** [comment]
Is there an issue/branch we can track to be notified when this is addressed?
